CombinatorialGames/Game/Loopy/Outcome.lean

```diff
@@ -0,0 +1,232 @@
+  mpr ls ll := ll.rec (motive_2 := fun _ _ ↦ _) (fun x y hyx _ hy ⟨s, hx, hs⟩ ↦
+    have ⟨r, hr⟩ := hs x hx y hyx; hy r hr.1 ⟨s, hr.2, hs⟩) (fun _ _ ↦ id) ls
+
+theorem not_isLeftWin_iff_mem_rightStrategy {x : LGame} :
+    ¬ IsLeftWin x ↔ ∃ s, x ∈ s ∧ IsRightStrategy s where
+  mp h := ⟨_, h, isRightStrategy_not_isLeftWin⟩
+  mpr ls ll := ll.rec (motive_2 := fun _ _ ↦ _) (fun x y hyx _ hy ⟨s, hx, hs⟩ ↦
+    have ⟨r, hr⟩ := hs x hx y hyx; hy r hr.1 ⟨s, hr.2, hs⟩) (fun _ _ ↦ id) ls
+
+theorem not_isLeftWin_and_isLeftLoss {x : LGame} : ¬ (IsLeftWin x ∧ IsLeftLoss x) :=
+  fun ⟨w, l⟩ ↦ not_isLeftWin_iff_mem_rightStrategy.mpr ⟨_, l, isRightStrategy_isLeftLoss⟩ w
+
+theorem not_isRightWin_and_isRightLoss {x : LGame} : ¬ (IsRightWin x ∧ IsRightLoss x) :=
+  fun ⟨w, l⟩ ↦ not_isRightWin_iff_mem_leftStrategy.mpr ⟨_, l, isLeftStrategy_isRightLoss⟩ w
+
+/-- `IsLeftDraw x` means that left draws `x` going first. -/
+def IsLeftDraw (x : LGame) : Prop := ¬ IsLeftWin x ∧ ¬ IsLeftLoss x
+
+/-- `IsRightDraw x` means that right draws `x` going first. -/
+def IsRightDraw (x : LGame) : Prop := ¬ IsRightWin x ∧ ¬ IsRightLoss x
+
+/-- The three possible outcomes of a game. -/
+inductive Outcome : Type | win | draw | loss
+
+/-- `leftOutcome x` is the outcome of `x` with left going first. -/
+noncomputable def leftOutcome (x : LGame) : Outcome := by classical
+  exact if IsLeftWin x then .win else if IsLeftLoss x then .loss else .draw
+
+/-- `rightOutcome x` is the outcome of `x` with right going first. -/
+noncomputable def rightOutcome (x : LGame) : Outcome := by classical
+  exact if IsRightWin x then .win else if IsRightLoss x then .loss else .draw
+
+theorem leftOutcome_eq_win_iff {x : LGame} : leftOutcome x = .win ↔ IsLeftWin x := by
+  classical rw [leftOutcome, Ne.ite_eq_left_iff]
+  split_ifs <;> rintro ⟨_⟩
+
+theorem rightOutcome_eq_win_iff {x : LGame} : rightOutcome x = .win ↔ IsRightWin x := by
+  classical rw [rightOutcome, Ne.ite_eq_left_iff]
+  split_ifs <;> rintro ⟨_⟩
+
+theorem leftOutcome_eq_loss_iff {x : LGame} : leftOutcome x = .loss ↔ IsLeftLoss x := by
+  classical rw [leftOutcome]
+  split_ifs with w l
+  · exact false_iff _ ▸ fun l ↦ not_isLeftWin_and_isLeftLoss ⟨w, l⟩
+  all_goals simpa
+
+theorem rightOutcome_eq_loss_iff {x : LGame} : rightOutcome x = .loss ↔ IsRightLoss x := by
+  classical rw [rightOutcome]
+  split_ifs with w l
+  · exact false_iff _ ▸ fun l ↦ not_isRightWin_and_isRightLoss ⟨w, l⟩
+  all_goals simpa
+
+theorem leftOutcome_eq_draw_iff {x : LGame} : leftOutcome x = .draw ↔ IsLeftDraw x := by
+  classical rw [leftOutcome]
+  split_ifs with w l
+  on_goal 3 => simpa using ⟨w, l⟩
+  all_goals rw [false_iff]
+  exacts [(·.1 w), (·.2 l)]
+
+theorem rightOutcome_eq_draw_iff {x : LGame} : rightOutcome x = .draw ↔ IsRightDraw x := by
+  classical rw [rightOutcome]
+  split_ifs with w l
+  on_goal 3 => simpa using ⟨w, l⟩
+  all_goals rw [false_iff]
+  exacts [(·.1 w), (·.2 l)]
+
+example (x : LGame) :
```

Comment:
I don't think we need these anymore, `not_isLeftWin_and_isLeftLoss` is a nicer way to state essentially the same.